### PR TITLE
Suppress stacktrace of InterruptedException in CommonCacheNotifier

### PR DESCRIPTION
**Pull request by kfaraz**
Addresses #10709

### Description

When `CommonCachedNotifier` is being stopped while the thread is waiting on `updateQueue.take()`,
an InterruptedException is thrown. The stack trace from this exception gives the wrong idea that something went wrong with the shutdown.

<hr>

This PR has:
- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

**Comment by abhishekagarwal87** [comment]
how does the `InterruptedException` affect graceful shutdown?

File: extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/CommonCacheNotifier.java [hunk 1]
```diff
@@ -140,6 +140,11 @@ public void start()
 
               LOG.debug(callerName + ":Received responses for cache update notifications.");
             }
+            catch (InterruptedException e) {
+              LOG.noStackTrace()
```
Comment (abhishekagarwal87) [inline extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/CommonCacheNotifier.java:144]: should this be emitted as an alert?

**Comment by kfaraz** [comment]
The stack trace from the InterruptedException might give the wrong notion that something went wrong with the shutdown. 

By graceful shutdown, I meant avoiding the above confusion. Let me fix this description 🙂

File: extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/CommonCacheNotifier.java (hunk 1 above)
Comment (kfaraz) [reply extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/CommonCacheNotifier.java:144]: I guess we could just log an INFO here instead.

Comment (abhishekagarwal87) [reply extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/CommonCacheNotifier.java:144]: makes sense to me.

**Comment by abhishekagarwal87** [comment]
> The stack trace from the InterruptedException might give the wrong notion that something went wrong with the shutdown.
> 
> By graceful shutdown, I meant avoiding the above confusion. Let me fix this description 🙂

Thanks 🙂

**Comment by abhishekagarwal87** [comment]
Ignoring code coverage check and merging since it is just a logging change.

**Review by abhishekagarwal87 — APPROVED** [review]

**Pull request merged**